ytypes/list_test.go

```diff
@@ -1712,6 +1712,153 @@ func TestSimpleMapKeyValueCreation(t *testing.T) {
+			keys: map[string]string{"key": "string-value"},
+			inSchema: &yang.Entry{
+				Name:     "key-list",
+				Kind:     yang.DirectoryEntry,
+				ListAttr: yang.NewDefaultListAttr(),
+				Key:      "key",
+				Config:   yang.TSTrue,
+				Dir: map[string]*yang.Entry{
+					"key": {
+						Kind: yang.LeafEntry,
+						Name: "key",
+						Type: &yang.YangType{Kind: yang.Yleafref, Path: "/cs:foo/cs:baz"},
+						Parent: &yang.Entry{
+							Name: "key",
+							Parent: &yang.Entry{
+								Name: "list-parent",
+								Parent: &yang.Entry{
+									Name: "root",
+									// This is the root.
+									Dir: map[string]*yang.Entry{
+										"foo": {
+											Dir: map[string]*yang.Entry{
+												"bar": {
+													Type: &yang.YangType{
+														Kind: yang.Ystring,
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			container:    &simpleStruct{KeyList: map[string]*ListStringStruct{}},
+			errSubstring: "cannot find leafref cs:baz",
+		},
+		{
+			desc: "absolute leafref used as key of list - but can't find top-level node",
+			keys: map[string]string{"key": "string-value"},
+			inSchema: &yang.Entry{
+				Name:     "key-list",
+				Kind:     yang.DirectoryEntry,
+				ListAttr: yang.NewDefaultListAttr(),
+				Key:      "key",
+				Config:   yang.TSTrue,
+				Dir: map[string]*yang.Entry{
+					"key": {
+						Kind: yang.LeafEntry,
+						Name: "key",
+						Type: &yang.YangType{Kind: yang.Yleafref, Path: "/cs:foo/cs:baz"},
+						Parent: &yang.Entry{
+							Name: "key",
+							Parent: &yang.Entry{
+								Name: "list-parent",
+								Parent: &yang.Entry{
+									Name: "root",
+									// This is the root.
+									Dir: map[string]*yang.Entry{
+										// "foo" is not specified here.
+										"bat": {
+											Dir: map[string]*yang.Entry{
+												"bar": {
+													Type: &yang.YangType{
+														Kind: yang.Ystring,
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			container:    &simpleStruct{KeyList: map[string]*ListStringStruct{}},
+			errSubstring: "can't find top-level foo",
+		},
+		{
+			desc: "absolute leafref used as key of list - but can't find top-level node",
+			keys: map[string]string{"key": "string-value"},
+			inSchema: &yang.Entry{
+				Name:     "key-list",
+				Kind:     yang.DirectoryEntry,
+				ListAttr: yang.NewDefaultListAttr(),
+				Key:      "key",
+				Config:   yang.TSTrue,
+				Dir: map[string]*yang.Entry{
+					"key": {
+						Kind: yang.LeafEntry,
+						Name: "key",
+						Type: &yang.YangType{Kind: yang.Yleafref, Path: "../baz"},
+						Parent: &yang.Entry{
+							Name: "key",
+						},
+					},
+				},
+			},
+			container:    &simpleStruct{KeyList: map[string]*ListStringStruct{}},
+			errSubstring: `cannot find leafref "../baz"`,
+		},
```

Comment:
Isn't this a relative leafref test? ../ goes up to the list level right?

Reply:
Yes, sorry, this was a copy-paste error as I was adding coverage for lines that weren't covered. Fixed the testcase name.